src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -83,11 +95,13 @@ public static function getSubscribedEvents(): array
 
     private function getIsGrantedString(IsGranted $isGranted): string
     {
-        $attributes = array_map(fn ($attribute) => '"'.$attribute.'"', (array) $isGranted->attributes);
+        $attributes = $isGranted->attributes instanceof Expression ? (string) $isGranted->attributes : $isGranted->attributes;
+        $attributes = array_map(fn ($attribute) => '"'.$attribute.'"', (array) $attributes);
```

Comment:
what about this (and similar for subject?)
```php

        $attributes = $isGranted->attributes instanceof Expression ? [$isGranted->attributes] : $isGranted->attributes;
        $attributes = array_map(fn ($attribute) => sprintf($attribute instanceof Expression? 'new Expression("%s")' : '"%s"', $attribute), (array) $attributes);
```

Reply:
Suggestion applied